app/controllers/api/v2/compute_resources_controller.rb

```diff
@@ -68,7 +68,11 @@ def change_datacenter_to_uuid(datacenter)
 
       def create
         begin
-          @compute_resource = ComputeResource.new_provider(compute_resource_params.except(:datacenter))
+          if compute_resource_params["provider"] == "Ovirt"
```

Comment:
Actually this should compare the `downcase`d version with 'ovirt', just to be sure.